lc3/codec/inc/LC3API.h

```diff
@@ -0,0 +1,441 @@
+ * to the total memory required and returns @ref LC3_RESULT_INSUFFICIENT_RESOURCES. The
+ * caller can then call this function again with a buffer of at least bufferSize's
+ * referenced value. A size of 0 will always trigger this condition. The buffer
+ * parameter may be NULL.
+ *
+ * <b>*bufferSize >= memory required by decoder:</b> Uses the user-provided buffer.
+ * bufferSize's referenced value is set to the amount of memory used by the decoder.
+ * The buffer parameter may not be NULL.
+ *
+ * @param[in]      sampleRate    Output PCM sample rate in Hz.
+ * @param[in]      bitsPerSample Output PCM bits per sample.
+ * @param[in]      frameSize     Frame size of 7.5 or 10 msec.
+ * @param[in]      buffer        Pointer to a memory buffer.
+ * @param[in, out] bufferSize    Size of the memory buffer passed in, number of bytes used returned.
+ * @param[out]     result        Pointer to an LC3 Return Code integer value.
+ *
+ * @returns  Non-NULL LC3DecoderHandle_t on success or NULL on failure.
+ */
+LC3DecoderHandle_t OSALCALL LC3DecodeSessionOpen(uint16_t sampleRate, uint8_t bitsPerSample, LC3FrameSize_t frameSize, uint8_t *buffer, uint16_t *bufferSize, int32_t *result);
+
+
+/**@brief Decodes a frame of data using LC3.
+ *
+ * Processes the frame of encoded bit stream data passed by the decodeInput structure.
+ * Returns the decoded audio PCM data via the decodeOutput structure.
+ *
+ * This function will automatically apply packet loss concealment (PLC) to any frame where
+ * the decode input's badFrameIndicator is set to BadFrame or if the bit stream data contains
+ * errors. The caller should set badFrameIndicator to BadFrame if the caller knows in advance
+ * of bit stream errors; this allows the decoder to skip decoding  and directly apply PLC. If
+ * the caller is unaware of bit stream errors, but they are present, the decoder will detect
+ * them regardless. When PLC is applied successfully, this function still returns
+ * @ref LC3_RESULT_NO_ERROR as when a frame is decoded successfully, but the decode output's
+ * PLCCounter parameter is incremented. Upon a successful decode, PLCCounter is reset.
+ * Therefore, a non-zero PLCCounter indicates PLC has been applied to the current frame.
+ *
+ * @param[in] decodeHandle  Handle to a decoder instance.
+ * @param[in] decodeInput   Pointer to a structure pointing to the encoded bit stream frame
+ *                          to decode.
+ * @param[in] decodeOutput  Pointer to a structure to receive the frame's decoded PCM data.
+ *
+ * @returns  Zero on success or one of the defined @ref LC3ResultCodes "LC3 result codes" on error.
+ */
+int32_t OSALCALL LC3DecodeSessionData(LC3DecoderHandle_t decodeHandle, LC3DecodeInput_t *decodeInput, LC3DecodeOutput_t *decodeOutput);
+
+
+/**@brief Closes an LC3 Decoder session.
+ *
+ * This function closes the decodeHandle's session and and releases allocated session memory.
```

Comment:
```suggestion
 * This function closes the decodeHandle's session and releases allocated session memory.
```

Reply:
Cannot edit this file due to legal reasons.